src/artemis/ispyb/store_in_ispyb.py

```diff
@@ -0,0 +1,133 @@
+import ispyb
+from sqlalchemy.connectors import Connector
+
+from src.artemis.ispyb.ispyb_dataclasses import GridInfo, DataCollection, DataCollectionGroup, Position
+
+
+ISPYB_CONFIG_FILE = "/dls_sw/dasc/mariadb/credentials/ispyb-dev.cfg"
+
+
+class StoreInIspyb:
+
+    conn: Connector
+
+    def store_grid_scan(self, grid_info: GridInfo, data_collection: DataCollection, position: Position,
+                        data_collection_group: DataCollectionGroup):
+
+        with ispyb.open(ISPYB_CONFIG_FILE) as self.conn:
+            data_collection.data_collection_group_id = self.store_data_collection_group_table(data_collection_group)
+            data_collection.position_id = self.store_position_table(position)
+
+            data_collection_id = self.store_data_collection_table(data_collection)
+
+            grid_info.ispyb_data_collection_id = data_collection_id
+
+            grid_id = self.store_grid_info_table(grid_info)
+
+            return grid_id, data_collection_id
+
+    def update_grid_scan_with_end_time_and_status(self, end_time: str, run_status: str, dc_id: int) -> int:
+        with ispyb.open(ISPYB_CONFIG_FILE) as self.conn:
+            mx_acquisition = self.conn.mx_acquisition
+
+            params = mx_acquisition.get_data_collection_params()
+            params["id"] = dc_id
+            params["endtime"] = end_time
+            params["run_status"] = run_status
+
+            return mx_acquisition.upsert_data_collection(list(params.values()))
+
+    def store_grid_info_table(self, grid_info: GridInfo) -> int:
+        mx_acquisition = self.conn.mx_acquisition
+
+        params = mx_acquisition.get_dc_grid_params()
+
+        params["parentid"] = grid_info.ispyb_data_collection_id
+        params["dxInMm"] = grid_info.dx_mm
+        params["dyInMm"] = grid_info.dy_mm
+        params["stepsX"] = grid_info.steps_x
+        params["stepsY"] = grid_info.steps_y
+        params["pixelsPerMicronX"] = grid_info.pixels_per_micron_x
+        params["pixelsPerMicronY"] = grid_info.pixels_per_micron_y
+        params["snapshotOffsetXPixel"] = grid_info.snapshot_offset_pixel_x
+        params["snapshotOffsetYPixel"] = grid_info.snapshot_offset_pixel_y
+        params["orientation"] = grid_info.orientation
+        params["snaked"] = grid_info.snaked
+
+        return mx_acquisition.upsert_dc_grid(list(params.values()))
+
+    def store_data_collection_table(self, data_collection: DataCollection) -> int:
+        core = self.conn.core
+        mx_acquisition = self.conn.mx_acquisition
```

Comment:
If we move core and mx_acquisition into instance variables we avoid a bit of code duplication